### PR TITLE
add more detailed documentation for sso in readthedocs

### PR DESCRIPTION
**Pull request by biyeun**
## Summary
Documentation update for SSO.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story
Just a documentation update. Built the docs locally to ensure I didn't break anything

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: docs/sso/architecture.rst
Comment (gherceg) [inline docs/sso/architecture.rst:5]:
```suggestion
    Everything related to SSO on CommCare HQ can be found in `corehq.apps.sso`
```

File: docs/sso/architecture.rst
Comment (gherceg) [inline docs/sso/architecture.rst:66]:
```suggestion
the `IdentityProvider` is misconfigured or the certificate expires and a user
```

File: docs/sso/architecture.rst
Comment (gherceg) [inline docs/sso/architecture.rst:84]:
```suggestion
Once trust is established, any user authenticating with that `IdentityProvider`
```
I could go either way here 🤷

File: docs/sso/local_setup.rst
Comment (gherceg) [inline docs/sso/local_setup.rst:25]: I wonder if it is worth moving this `Create an Enterprise Software Plan` section to its own section and referencing it here, as we have found a need for this in other areas outside of SSO right? And having it in a specific location for enterprise setup would make it easier to maintain when, for example, we update the enterprise software plan that should be referenced.

File: docs/sso/overview.rst
Comment (gherceg) [inline docs/sso/overview.rst:11]:
```suggestion
    Pay attention to the difference between **authentication** and **authorization**.
```

File: docs/sso/overview.rst
Comment (gherceg) [inline docs/sso/overview.rst:15]:
```suggestion
    **Authorization** is the set of permissions given to an authenticated user
```
Saying "on a system" is probably only needed once in this sentence, fine removing the other one instead if that sounds better.

File: docs/sso/overview.rst
Comment (gherceg) [inline docs/sso/overview.rst:43]:
```suggestion
To create this trust, x509 certificates are exchanged between the IdP and the SP.
```

**Review by gherceg — APPROVED** [review]
Just left some suggestions around wording, but looks good.

File: docs/sso/local_setup.rst
Comment (biyeun) [reply docs/sso/local_setup.rst:25]: maybe a Local Setup sub section is needed (outside of SSO)?

Comment (gherceg) [reply docs/sso/local_setup.rst:25]: Yeah that sounds reasonable. Doesn't have to block this PR,  you or I could move it over in a separate PR after this is merged

File: docs/sso/overview.rst
Comment (joxl) [inline docs/sso/overview.rst:15]:
```suggestion
    **Authorization** is the process used to determine the set of permissions to grant a user
```

**Review by joxl — APPROVED** [review]

**Pull request merged**
